### ISSUE TITLE
Crash in topdown

### ISSUE THREAD

**Issue by acesnik** [smith-chem-wisc/MetaMorpheus#1643]
This happened in fraction 12 of the jurkat topdown dataset. It looks like there's some bug around collecting MS2 scans... I know fxn 12's are kind of weird, so maybe it has something to do with the chemistry of the fraction.

```
MetaMorpheus: version Not a release version. DEBUG.
Data files were processed on a Dell Inc. Precision Tower 7910 computer using Windows version 6.2.9200.0 with a 64-Bit processor and 24 cores operating at 2.993GHz and 128GB installed RAM.
.NET Framework Version: 4.7 or laterThermo MSFileReader Fileio DLL		Version: 3.1
C:\Program Files\Thermo\MSFileReader\fregistry_x64.dll MISSING\nThermo MSFileReader Rawfile Com Interface		Version: 3.1.9.0

e: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: toExclusive
   at System.Collections.Concurrent.Partitioner.Create(Int32 fromInclusive, Int32 toExclusive)
   at TaskLayer.MetaMorpheusTask.GetMs2Scans(MsDataFile myMSDataFile, String fullFilePath, CommonParameters commonParameters) in E:\source\repos\MetaMorpheus\TaskLayer\MetaMorpheusTask.cs:line 97
   at TaskLayer.SearchTask.RunSpecific(String OutputFolder, List`1 dbFilenameList, List`1 currentRawFileList, String taskId, FileSpecificParameters[] fileSettingsList) in E:\source\repos\MetaMorpheus\TaskLayer\SearchTask\SearchTask.cs:line 171
   at TaskLayer.MetaMorpheusTask.RunTask(String output_folder, List`1 currentProteinDbFilenameList, List`1 currentRawDataFilepathList, String displayName) in E:\source\repos\MetaMorpheus\TaskLayer\MetaMorpheusTask.cs:line 377e.Message: Specified argument was out of the range of valid values.
Parameter name: toExclusivee.InnerException: e.Source: mscorlibe.StackTrace:    at System.Collections.Concurrent.Partitioner.Create(Int32 fromInclusive, Int32 toExclusive)
   at TaskLayer.MetaMorpheusTask.GetMs2Scans(MsDataFile myMSDataFile, String fullFilePath, CommonParameters commonParameters) in E:\source\repos\MetaMorpheus\TaskLayer\MetaMorpheusTask.cs:line 97
   at TaskLayer.SearchTask.RunSpecific(String OutputFolder, List`1 dbFilenameList, List`1 currentRawFileList, String taskId, FileSpecificParameters[] fileSettingsList) in E:\source\repos\MetaMorpheus\TaskLayer\SearchTask\SearchTask.cs:line 171
   at TaskLayer.MetaMorpheusTask.RunTask(String output_folder, List`1 currentProteinDbFilenameList, List`1 currentRawDataFilepathList, String displayName) in E:\source\repos\MetaMorpheus\TaskLayer\MetaMorpheusTask.cs:line 377e.TargetSite: System.Collections.Concurrent.OrderablePartitioner`1[System.Tuple`2[System.Int32,System.Int32]] Create(Int32, Int32)
```

**Comment by rmillikin** [comment]
kind of weird. seems like there aren't any MS2 scans? can you share this file?

**Comment by acesnik** [comment]
Search the two fraction 12 files in `//bison/share/Users/Sean/Jurkat Proteoform Data/Jurkat Top-Down Raw and HitReport`. I'm on my phone so that might not be the exact path.

**Comment by acesnik** [comment]
Specifically, the offending file is located at `"\\bison\share\Users\Sean\Jurkat Proteoform Data\Jurkat Top-Down Raw and HitReport\FXN12_tr1_031817.raw"`

**Comment by rmillikin** [comment]
As far as I can tell, this file has no MS2 scans; I looked via mzLib code and QualBrowser. I'll fix the crash in the case of no MS2 scans, though.

**Comment by rmillikin** [comment]
fixed in #1655
